atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/ServerImplementationTest.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.atlasdb.timelock.config.TimeLockServerConfiguration;
+
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+import io.dropwizard.lifecycle.setup.LifecycleEnvironment;
+import io.dropwizard.setup.Environment;
+
+public abstract class ServerImplementationTest {
```

Comment:
rename to `TimelockServerTest` and make it non-abstract. This test should verify that ``TimeLockServer#run`` , or `TimeLockServer#stop` etc do the right thing(calls the appropriate methods from server implementation) always. The `TimeLockServerConfiguration ` can be mocked too as it should work irrespective of the config type. 

We should have an `AbstractServerImplementationTest` class that can be extended by `AtomixServerImplementationTest` etc, this should verify that ServerImplementation methods do the right thing. So the config should be pluggable in this class, and there can be tests like 

```
public void verifyPortsAreAvailableOnStartUp() {
  serverImpl.onStartup(configuration);
  tryToConnectToTimelockPort();
}
```